source/readonly-keys-of.d.ts

```diff
@@ -0,0 +1,37 @@
+/**
+Extract all readonly keys from the given type.
+
+This is useful when you want to create a new type that contains readonly keys only.
+
+@example
+```
+import type {ReadonlyKeysOf, Except} from 'type-fest';
+
+interface User {
+	name: string;
+	surname: string;
+	readonly id: number;
+}
+
+type UpdateResponse<Entity extends object> = Pick<Entity, ReadonlyKeysOf<Entity>>;
+
+const update1: UpdateResponse<User> = {
+    id: 123,
+};
+```
+
+@category Utilities
+*/
+// https://github.com/Microsoft/TypeScript/issues/27024#issuecomment-421529650
+type IfEquals<X, Y, A, B> =
+	(<T>() => T extends X ? 1 : 2) extends
+	(<T>() => T extends Y ? 1 : 2) ? A : B;
```

Comment:
Use https://github.com/sindresorhus/type-fest/blob/bba240fb01ac0b540cb8f163d17854046fa08905/source/is-equal.d.ts#L12 

4 type parameters are pretty unreadable. Better to use extends:

```
IsEqual<Value[0], Item> extends true
			? true
			: Includes<rest, Item>
```